evennia/utils/evmenu.py

```diff
@@ -1014,7 +1149,7 @@ def display_helptext(self):
 
     def nodetext_formatter(self, nodetext):
         """
-        Format the node text itself.
+        Format the node text itself. Legacy code. Use is deprecated.
```

Comment:
Thinking aloud: I wonder if it may be an idea to log a deprecationwarning for this format, to make people convert their existing menus. Maintaining both forms makes EvMenu considerably more complex ...

Reply:
I could certainly add something to log that a deprecated function is being used, but that seems to me to just add another complication to the code rather than removing anything.

At present I think we should continue to support both formats because we don't want to make everyone have to go back and rewrite all their menus. If we do eventually decide that we want to remove the original functions and force rewrites we should probably do that as a sort of phased rollout. Phase 1 is both formats continue to work equally with no warnings, just things in the docstring and tutorials saying this is the new preferred format. Phase 2 would be where we actually start to give gentle nudges with log warnings and the like. That way if people say 'why am I suddenly getting these warnings' the reply can be 'we said six months ago that we plan to deprecate the old method'. Phase 3 would be the final removal of the code.

Reply:
Fair enough. We can deprecate later. It's just that the documentation gets ever more complex.